wappsto/connection/message_log.py

```diff
@@ -0,0 +1,304 @@
+        now = datetime.datetime.now()
+        if self.compression_period >= MONTH_PERIOD:
+            file_name = str(now.year) + "-" + str(now.month)
+        if self.compression_period >= DAY_PERIOD:
+            file_name += "-" + str(now.day)
+        if self.compression_period >= HOUR_PERIOD:
+            file_name += "-" + str(now.hour)
+        file_name += ".txt"
+        return file_name
+
+    def get_logs(self):
+        """
+        Gets log files in the location.
+
+        Gets all files from directory and return ones that follow log file format.
+
+        Returns:
+            list of log file names.
+        """
+        file_list = os.listdir(self.log_location)
+        pattern = "[0-9][0-9][0-9][0-9]-([0-9]|1[0-2])"
+        return [file_name for id, file_name in enumerate(file_list) if re.search(pattern, file_name)]
+
+    def compact_logs(self):
+        """
+        Compacts all logs to save space.
+
+        Uses all logs received from "get_logs" method and compacts the ones that
+        are of type text (after compacting the text file is deleted).
+        """
+        all_logs = self.get_logs()
+        text_logs = [file_name for id, file_name in enumerate(all_logs) if re.search(".txt$", file_name)]
+        for file_name in text_logs:
+            file_path = self.get_file_path(file_name)
+            zip_file = zipfile.ZipFile(file_path.replace(".txt", ".zip"), "w")
+            zip_file.write(file_path, file_name)
+            zip_file.close()
+            os.remove(file_path)
+
+    def get_oldest_log_name(self):
+        """
+        Gets the oldest log's name.
+
+        Uses all logs received from "get_logs" method and sorts them, taking the first (oldest).
+
+        Returns:
+            name of the file.
+        """
+        all_logs = self.get_logs()
+        all_logs.sort()
+        file_name = all_logs[0]
+        return file_name
+
+    def get_text_log(self, file_name):
+        """
+        Gets name of the file.
+
+        Checks if the file is compacted, if it is then it is unzipped and new name is returned,
+        otherwise same file is returned.
+
+        Args:
+            file_name: name of the file.
+
+        Returns:
+            name of the file.
+        """
+        if re.search(".zip$", file_name):
+            file_path = self.get_file_path(file_name)
+            with zipfile.ZipFile(file_path, "r") as zip_file:
+                zip_file.extractall(self.log_location)
+            os.remove(file_path)
+            file_name = file_name.replace(".zip", ".txt")
+        return file_name
+
+    def remove_data(self, file_name):
+        """
+        Removes data.
+
+        Removes lines from the file, but if the number of lines to remove
+        exceeds lines in file or file is not text format the file is deleted.
+
+        Args:
+            file_name: name of the file.
+
+        """
+        file_path = self.get_file_path(file_name)
+        if re.search(".txt$", file_name):
```

Comment:
Change so that it delete the file and returns here, and move the debug message to the beginning.